src/deepsparse/benchmark_model/benchmark_model.py

```diff
@@ -0,0 +1,251 @@
+        "model_path",
+        type=str,
+        help="Path to an ONNX model file or SparseZoo model stub",
+    )
+
+    parser.add_argument(
+        "-b",
+        "--batch_size",
+        type=int,
+        default=1,
+        help="The batch size to run the analysis for. Must be greater than 0",
+    )
+    parser.add_argument(
+        "-shapes",
+        "--input_shapes",
+        type=str,
+        default="",
+        help="Override the shapes of the inputs, "
+        'i.e. -shapes "[1,2,3],[4,5,6],[7,8,9]" results in '
+        "input0=[1,2,3] input1=[4,5,6] input2=[7,8,9]",
+    )
+    parser.add_argument(
+        "-ncores",
+        "--num_cores",
+        type=int,
+        default=None,
+        help=(
+            "The number of physical cores to run the analysis on, "
+            "defaults to all physical cores available on the system"
+        ),
+    )
+    parser.add_argument(
+        "-s",
+        "--scenario",
+        type=str,
+        default="async",
+        choices=["async", "sync"],
+        help=(
+            "Choose between using a sync/async scenarios. This is similar to the "
+            "single-stream/multi-stream scenarios. Default value is async."
+        ),
+    )
+    parser.add_argument(
+        "-t",
+        "--time",
+        type=int,
+        default=10,
+        help="The number of seconds the benchmark will run. Default is 10 seconds.",
+    )
+    parser.add_argument(
+        "-nstreams",
+        "--num_streams",
+        type=int,
+        default=None,
+        help=(
+            "The number of streams that will submit inferences in parallel using "
+            "async scenario. Default is automatically determined for given hardware "
+            "and may be sub-optimal."
+        ),
+    )
+    parser.add_argument(
+        "-pin",
+        "--thread_pinning",
+        type=str,
+        default="core",
+        choices=["none", "core", "numa"],
+        help=(
+            "Enable binding threads to cores ('core' the default), "
+            "threads to cores on sockets ('numa'), or disable ('none')"
+        ),
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        help="Lower logging verbosity",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "-x",
+        "--export_path",
+        help="Store results into a JSON file",
+        type=str,
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def decide_thread_pinning(pinning_mode: str):
+    pinning_mode = pinning_mode.lower()
+
+    if pinning_mode in "core":
+        os.environ["NM_BIND_THREADS_TO_CORES"] = "1"
+        log.info("Thread pinning to cores enabled")
+    elif pinning_mode in "none":
+        os.environ["NM_BIND_THREADS_TO_CORES"] = "0"
+        log.info("Thread pinning to cores disabled, performance may be sub-optimal")
+    elif pinning_mode in "numa":
+        os.environ["NM_BIND_THREADS_TO_SOCKETS"] = "1"
```

Comment:
maybe `os.environ["NM_BIND_THREADS_TO_CORES"] = "0"` just to be extra paranoid here?